include/heuristic/HeuristicMapper.hpp

```diff
@@ -263,6 +269,34 @@ class HeuristicMapper : public Mapper {
     }
     return currentCost + newCost;
   }
+
+  static double computeEffectiveBranchingRate(std::size_t       nodesProcessed,
+                                              const std::size_t solutionDepth) {
+    // N = (b*)^d + (b*)^(d-1) + ... + (b*)^2 + b* + 1
+    // no closed-form solution for b*, so we use approximation via binary search
+    if (solutionDepth == 0) {
+      return 0.;
+    }
+    --nodesProcessed; // N - 1 = (b*)^d + (b*)^(d-1) + ... + (b*)^2 + b*
+    double lower = std::pow(static_cast<double>(nodesProcessed),
+                            1.0 / static_cast<double>(solutionDepth)) /
+                   static_cast<double>(solutionDepth);
+    double upper = std::pow(static_cast<double>(nodesProcessed),
+                            1.0 / static_cast<double>(solutionDepth));
+    while (upper - lower > 2 * EFFECTIVE_BRANCH_RATE_TOLERANCE) {
+      const double mid = (lower + upper) / 2.0;
+      double       sum = 0.0;
+      for (std::size_t i = 1; i <= solutionDepth; ++i) {
+        sum += std::pow(mid, i);
+      }
+      if (sum < static_cast<double>(nodesProcessed)) {
+        lower = mid;
+      } else {
+        upper = mid;
+      }
+    }
+    return (lower + upper) / 2.0;
```

Comment:
Well, yes, the return value would still be valid in that case, but it would need to perform 1 more iteration. Since the true solution is always guaranteed to be between `lower` and `upper`, the best guess is always the mid-point between those two. Returning one of the boundary values would be just throwing away half the precision, which could be compensated by another iteration, but I see no reason to do that.

Reply:
Another good point. In that case, I am totally fine with the solution here.